tests/meshcop/commissioner.cpp

```diff
@@ -705,11 +864,30 @@ int run(Context &context)
     Coap::Resource relayReceiveHandler(OT_URI_PATH_RELAY_RX, HandleRelayReceive, &context);
     Coap::Resource joinerFinalizeHandler(OT_URI_PATH_JOINER_FINALIZE, HandleJoinerFinalize, &context);
 
+    otbrLog(OTBR_LOG_INFO, "agent-address: %s", context.mAgent.mAddress_ascii);
+    if (context.mAgent.mAddress_ascii[0] == 0)
+    {
+        fail("Missing AGENT ip address\n");
+    }
+
+    otbrLog(OTBR_LOG_INFO, "agent-port: %s", context.mAgent.mPort_ascii);
+    if (context.mAgent.mPort_ascii[0] == 0)
+    {
+        fail("Missing AGENT ip port\n");
+    }
+
+    if (context.mJoiner.mPSKd_ascii[0] == 0)
+    {
+        fail("Missing PSKd (joiner passphrase/password)\n");
+    }
+
     context.mSsl = &ssl;
     context.mNet = &client_fd;
-#if defined(MBEDTLS_DEBUG_C)
-    mbedtls_debug_set_threshold(DEBUG_LEVEL);
-#endif
+    {
+        int n;
+        n = otbrLogGetLevel();
+        mbedtls_debug_set_threshold(n);
```

Comment:
could be but why should it?

Of the places I have worked over the last 30+ years writing C code I have seen multiple coding standards across multiple companies.  What you do greatly depends on what your measuring stick is, and how you judge code.

if it is "code coverage" - you write code that is very hard to debug.

If it is "cost to maintain" - you have look at what some coverage rules do to the debug process. Remember: many others will step through this code at some point in the future - and that cost is higher then you think.

thus there are 3 reasons one writes code this way:

1) In a debugger you can stop/halt - and inspect a variable, or modify it during debugging.

This is very important if what looks like a function is actually a macro... 

2) In a debugger a statement like  `foo( bar() )`  is really hard to step through. Debuggers only step *LINES* - not columns. This example is simple: there is a *SINGLE* nested function but there are many cases where there are more complex statements that make it super hard.  

Another is that of practicality - often various GUI wrappers for GDB can't always step correctly - then you have lost your debug state, this includes Eclipse, Emacs-GUD mode, and countless others.

3) While it *looks* more efficient, let the optimizer make it better.  it can always do a better job then you.